api/query/atoms_test.go

```diff
@@ -616,6 +674,54 @@ func TestStructuredQuery_bindCount(t *testing.T) {
 	assert.Equal(t, expected, q.BindToRuns(runs...))
 }
 
+func tearDownLinkSubTest() func(t *testing.T, url string) {
+	return func(t *testing.T, url string) {
```

Comment:
This is a bit convoluted. You don't need to return a functor to capture `url`. Parameters are captured at the `defer` statement. e.g. https://play.golang.org/p/3znsrNkGFB_F

And then you can simply do `defer tearDownLinkSubTest(t, shared.MetadataArchiveURL)`